GameEngine/GameEngine.h

```diff
@@ -45,7 +46,7 @@ class GameState {
 };
 
 // The GameEngine class manages the game states and processes commands.
-class GameEngine {
+class GameEngine : public Map{
```

Comment:
um what? why r u making the gameengine a child of the map class

Reply:
it's supposed to be ILoggable and Subject, idk where that went.